src/btree/bt_ret.c

```diff
@@ -352,20 +264,30 @@ __wt_key_return(WT_CURSOR_BTREE *cbt)
 
 /*
  * __wt_value_return --
- *     Change the cursor to reference an internal return value.
+ *     Change the cursor to reference an update return value.
  */
 int
-__wt_value_return(WT_CURSOR_BTREE *cbt, WT_UPDATE *upd)
+__wt_value_return(WT_CURSOR_BTREE *cbt, WT_UPDATE_VALUE *upd_value)
 {
     WT_CURSOR *cursor;
+    WT_SESSION_IMPL *session;
 
     cursor = &cbt->iface;
+    session = (WT_SESSION_IMPL *)cbt->iface.session;
 
     F_CLR(cursor, WT_CURSTD_VALUE_EXT);
-    if (upd == NULL)
+    if (upd_value->type == WT_UPDATE_INVALID)
         WT_RET(__value_return(cbt));
```

Comment:
Did you see my comment about this? If you prefer to keep __value_return, can we add a FIXME here to indicate this is kept for fixed length column store?

Reply:
Yeah, I saw your comment. Didn't you say it's required for fixed-length column store? So I don't think I could remove it even if I wanted to.

Sure, I'll add a FIXME.

Reply:
I'm not sure how we are going to fix fixed length column store. But this should not be required if it is brought in line with durable history. I would just say, fixed length column store doesn't work currently but we need to keep this to pass the test. We will revisit this when fixing fixed length column store. I think it's also worth to change the code so that we only call this for fixed length column store. That can cut some confusion as well.

Reply:
62d46ab